### PR TITLE
inital nixos-module

### LINKED ISSUE #3: create nixos-module

### LINKED ISSUE #6: implement data sending for borzoi

### PR DESCRIPTION
**Pull request by marenz2569**
Closes #3 Closes #6

### CODE REVIEW

File: nixos-modules/default.nix [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      type = types.str;
+      default = "tetra-decoder";
+      description = "systemd user";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "tetra-decoder";
+      description = "group of systemd user";
+    };
+    instances = mkOption {
```
Comment (revol-xut) [inline nixos-modules/default.nix:22]: maybe confusing name:


```nix
borzoi = {
   url = "borzoi.tlm.solutions";
};
decoder = {
    receivePort = 666;
    transmittPort = 667; # btw can we not just use one socket ? 
    extraArgs = "--retarded";
};
config = {
    stationUUID = "000-000..."
};

```

File: nixos-modules/default.nix [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        serviceConfig = {
+          Type = "forking";
+          User = cfg.user;
+          Restart = "on-failure";
+          StartLimitBurst = "2";
+          StartLimitIntervalSec = "150s";
+        };
+      };
+
+      "tetra-sender-${instanceName}" = {
```
Comment (revol-xut) [inline nixos-modules/default.nix:88]: I dont like but I will make it beautiful some day TM

File: include/decoder.hpp [hunk 3]
```diff
@@ -7,14 +7,15 @@
  *   Tassilo Tanneberger
  */
 
-#ifndef TETRA_DECODER_DECODER_HPP
-#define TETRA_DECODER_DECODER_HPP
+#pragma once
```
Comment (tanneberger) [inline include/decoder.hpp:10]: not standard include guards?

File: include/decoder.hpp [hunk 4]
```diff
@@ -7,14 +7,15 @@
...
 
 #include <memory>
 #include <optional>
 #include <string>
 #include <vector>
 
+#include <bit_stream_decoder.hpp>
```
Comment (tanneberger) [inline include/decoder.hpp:17]: why not `#include "bit_stream_decoder.hpp"` ?

I think this is misleading and breaks if the cmake statement is removed.

File: include/iq_stream_decoder.hpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    const float SEQUENCE_DETECTION_THRESHOLD = 1.5;
+
+    std::shared_ptr<LowerMac> lower_mac_{};
+    std::shared_ptr<BitStreamDecoder> bit_stream_decoder_{};
+
+    bool is_uplink_{};
+
+    std::shared_ptr<StreamingOrderedOutputThreadPoolExecutor<std::vector<std::function<void()>>>> threadPool_;
+
+    std::thread upperMacWorkerThread_;
```
Comment (tanneberger) [inline include/iq_stream_decoder.hpp:77]: all the member variables are sneak case this is not

File: include/decoder.hpp (hunk 3 above)
Comment (marenz2569) [reply include/decoder.hpp:10]: as discussed we now use `#pragma once` everywhere

**Review by tanneberger — APPROVED** [review]
UwU very nice pullrequest kawaii

**Review by tanneberger — APPROVED** [review]

**Pull request merged**